src/main/java/core/basesyntax/impl/BalanceHandlerImpl.java

```diff
@@ -0,0 +1,15 @@
+package core.basesyntax.impl;
+
+import dao.FruitDao;
+import dao.FruitDaoImpl;
+import model.FruitTransaction;
+import strategy.OperationHandler;
+
+public class BalanceHandlerImpl implements OperationHandler {
+    private FruitDao fruitDao = new FruitDaoImpl();
```

Comment:
The `fruitDao` should not be initialized directly in the `BalanceHandlerImpl`. Instead, it should be passed as a parameter in the constructor to follow the Dependency Inversion principle, one of the SOLID principles. This will make your service more independent and testable.